server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -375,6 +375,14 @@ public void flush(boolean force, boolean waitIfOngoing) throws EngineException {
     @Override
     public void forceMerge(boolean flush, int maxNumSegments, boolean onlyExpungeDeletes,
                            boolean upgrade, boolean upgradeOnlyAncientSegments, String forceMergeUUID) {
+        if (maxNumSegments < lastCommittedSegmentInfos.size()) {
```

Comment:
This rejects force-merge requests that do not specify the `maxNumSegments` parameter at all, since that comes through to here as `maxNumSegments == ForceMergeSegments.Defaults.MAX_NUM_SEGMENTS == -1`. We should accept these requests too, it's ok for them to do nothing.

Reply:
That's right, I have updated the change.